File: src/exercise03/java/ContaPoupanca.java
package exercise03.java;

public class ContaPoupanca extends Conta {
    private double taxaDeRendimentoMensal = 1.05;

    public ContaPoupanca(Cliente titular) {
        super(titular);
    }

    public ContaPoupanca(double saldo, Cliente titular) {
        super(saldo, titular);
    }

    public ContaPoupanca(double saldo, Cliente titular, double taxaDeRendimentoMensal) {
        super(saldo, titular);
        this.taxaDeRendimentoMensal = 1.0 + taxaDeRendimentoMensal;
    }

    public double recolherJurosMensal(){
        double saldoFinal = (consultarSaldo() * taxaDeRendimentoMensal);
        double redimento = (saldoFinal - consultarSaldo());
        System.out.println("Recolhendo Juros. Saldo: " + redimento);
        return redimento;
    }
}

File: src/exercise03/kotlin/ContaPoupanca.kt
package exercise03.kotlin

class ContaPoupanca : Conta {
    private var taxaDeRendimentoMensal = 1.05

    constructor(titular: Cliente) : super(titular) {}
    constructor(saldo: Double, titular: Cliente) : super(saldo, titular) {}
    constructor(saldo: Double, titular: Cliente, taxaDeRendimentoMensal: Double) : super(saldo, titular) {
        this.taxaDeRendimentoMensal = 1.0 + taxaDeRendimentoMensal
    }

    fun recolherJurosMensal(): Double {
        val saldoFinal = consultarSaldo() * taxaDeRendimentoMensal
        val redimento = saldoFinal - consultarSaldo()
        println("Recolhendo Juros. Saldo: $redimento")
        return redimento
    }
}